dangvuthang/rocodingtest
# Feat/dynamic exam page

### Changelog

- Add dynamic page for **exam**
- Refactor **exam** to its own component
- Implement logic to ask for webcam permission, and exam validation
- Display time dynamically in accordance with `endDate` field
- Integrate with backend to handle  test `permission`, capture `evidence` when user cheat, and display compiling message
- Automatically submit when user cheat is equal to 3 


### How to test

- **Checkout** this branch
- Run both `frontend` and `backend`
- Navigate to `localhost:3000/exam/625d4e2ca9d3d42737bd46b4` and ensure that the data displayed is correct
- Navigate to `localhost:3000/exam/123` and ensure that no exam is available and the error message displayed correctly
- Try to click on `Run code` and check if the message display is correct
- Try to cheat 3 times and see if the submission is forced
- Check mongodb hackermit - records to see if a link to evidence is properly captured
